Jint/Native/Json/JsonParser.cs

```diff
@@ -1,90 +1,124 @@
+using System.Buffers;
+using System.Diagnostics.CodeAnalysis;
 using System.Globalization;
 using System.Runtime.CompilerServices;
+using System.Text;
 using Esprima;
-using Esprima.Ast;
 using Jint.Native.Object;
 using Jint.Pooling;
 using Jint.Runtime;
 
-using Range = Esprima.Range;
-
 namespace Jint.Native.Json
 {
     public sealed class JsonParser
     {
+        private static int s_defaultMaxDepth = 64;
         private readonly Engine _engine;
+        private readonly int _maxDepth;
+
+        /// <summary>
+        /// The max. depth level used for new JsonParser instances.
+        /// The initial value is 64.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">Is thrown if the provided value is negative.</exception>
+        public static int DefaultMaxDepth
```

Comment:
I don't think that's a common pattern to have. Maybe that should be something moved to the engine options.

Reply:
Can do that, no problem. Would `ConstraintOptions` be the right place for that?